e2e/test/provisioning/ReprovisioningE2ETests.cs

```diff
@@ -711,16 +726,14 @@ public async Task ProvisioningDeviceClient_ReprovisioningBlockingWorks_MqttWs_Sy
 
             if (provisioningAuth is AuthenticationProviderX509 x509Auth)
             {
-                X509Certificate2 cert = x509Auth.GetAuthenticationCertificate();
+                using X509Certificate2 cert = x509Auth.GetAuthenticationCertificate();
```

Comment:
If you put `using` here, it'll be disposed after line 731 and will not work when being used when referenced by `auth` in subsequent lines.

It has to be disposed in the `finally` block.